src/app/records/search-records.component.ts

```diff
@@ -39,33 +47,50 @@ export class SearchRecordsComponent
     this.type = this.route.snapshot.params.type;
 
     // Get search data
-    this.addSub(this.recordsService.getRecordDataForOwnerSearch({ owner: this.param, type: this.type }).subscribe(data => {
-      if (data.type.layout !== RecordLayoutEnum.LIST) {
-        throw new Error(`Invalid record layout: ${data.type.layout}`);
-      }
-
-      this.fieldsInSearch = data.customFields.filter(cf => data.fieldsInSearch.includes(cf.internalName));
-      this.fieldsInList = data.customFields.filter(cf => data.fieldsInList.includes(cf.internalName));
-      this.form.setControl('customValues', this.fieldHelper.customValuesFormGroup(this.fieldsInSearch, {
-        useDefaults: false
-      }));
-      this.form.patchValue(data.query);
-      this.data = data;
+    if (this.generalSearch) {
+      this.addSub(this.recordsService.getRecordDataForGeneralSearch({ type: this.type })
+        .subscribe(data => this.onInitialize(data)));
+    } else {
+      this.addSub(
+        this.recordsService.getRecordDataForOwnerSearch({ owner: this.param, type: this.type })
+          .subscribe(data => this.onInitialize(data)));
+    }
+  }
+
+  protected onInitialize(data: BaseRecordDataForSearch) {
```

Comment:
Why a separated method, when you could just set `this.data = data` and do everthing in `onDataInitialized(data)`?